servers/src/main/java/tachyon/master/next/filesystem/FileSystemMaster.java

```diff
@@ -388,14 +394,30 @@ public long getNewBlockIdForFile(long fileId) throws FileDoesNotExistException {
   public boolean deleteFile(long fileId, boolean recursive)
       throws TachyonException, FileDoesNotExistException {
     // TODO: metrics
+    long opTimeMs = System.currentTimeMillis();
+    boolean ret = deleteFileInternal(fileId, recursive, opTimeMs);
+    writeJournalEntry(new DeleteFileEntry(fileId, recursive, opTimeMs));
```

Comment:
The writing/flushing of the journal should be in the same critical section as the operation it is logging. So, that means it should be in the same synchronized block as the actual delete.